### ISSUE TITLE
Additional info to selected item as dropdown for added item

### ISSUE THREAD

**Issue by BoPeng** [crucialfelix/django-ajax-selects#262]
I am trying to create a field with users + permissions. Basically, when the query is executed, it should return  a list of `user + possible_permissions` like
```
(user, ['read', 'write', 'manage'])
```
Then the frontend should create a dropdown box to allow users select permission for the added user.

The field would return `user + permission` as either a tuple, or `permission` as an attribute to `user`.

Is this possible with `django-ajax-selects`?

**Comment by BoPeng** [comment]
It turns out to do "doable" but a lot of customizations are needed. The core part is that I need to "bundle" user + permission to a single object to be handled by `django-ajax-select` and write customized JS to change the permission of returned object.

**Comment by crucialfelix** [comment]
Hi, sorry for the delay!

Just to clarify further:

This library is for querying for models when you have way too many models to display in a select ui element. If you only had 3 models, then you could just use a normal Django admin select.

If you have a fixed list of strings such as "read" "write" "manage" then you can use a [ChoiceField](https://docs.djangoproject.com/en/3.0/ref/forms/fields/#choicefield).

If you are selecting permissions from the database (they are models) then you can use a [ModelChoiceField](https://docs.djangoproject.com/en/3.0/ref/forms/fields/#modelchoicefield) and this will work as long as you don't have too many.

So the only case where django-ajax-selects will help is if you are selecting models and there are a lot of them.

Hope that helps!

**Comment by BoPeng** [comment]
My use case is for users to search from the user database to choose collaborate, and set permissions to them once they select a collaborator, so the permission part is attached to user, and does not exist when the user is selected. I think this is more or less a common scenario with examples from dropbox or box.

My solution is ugly to say the best but it worked, and let me outline it here:

1. In the channel `get_query` returns a "combined object", which is something like
```
return [WithPermission(pk=-x.pk, user=x) for x in .... original_query]
```
here I use negative PK to indicate the user is new.

2. In the channel `format_item_display(self, item)` I display both the user and permission dropdown

```
return f'''<div...     name='permission_{item.pk}>'''
```
where the `name` part is very important.

3. Let `django_ajax_select` do its work. It will completely ignore the permission part, but I add remove users regardless of permission for now.

4. In the `view` part, I process the `POST` event 

```
        self.perms = {}
        for key in request.POST.keys():
            if key.startswith('permission_'):
                self.perms[int(key[11:])] = request.POST[key]
```
to get the value from the `name="permission_{item.pk}`, which is independent of the form, but available in the `POST` anyway.

From here I set permissions according to `pk`. Note that I messed with `add` and `delete` events with customized JS at first but chose to handle `POST` directly since it is much easier to do and more robust.
